ipa/ipa_provider.py

```diff
@@ -421,6 +423,19 @@ def _terminate_instance(self):
         """Terminate the instance."""
         raise NotImplementedError(NOT_IMPLEMENTED)
 
+    def _collect_vm_info(self):
```

Comment:
execute_ssh_command will raise an exception if the command fails. So per the discussion we need to either check that systemd exists before running these commands or put it all in a try catch.

My preference would be to use a try/except and log an info level message that vm info failed to collect.

Reply:
done